### PR TITLE
Refactor Fuse user group policy

### LINKED ISSUE #15822: Supports Fuse chown user or group separately
**Alluxio Version:**
2.9.0-SNAPSHOT

**Describe the bug**
Currently when chown user or group only, the non-set user or group is INVALID user/group id and chown will return 0 and do nothing which violate the POSIX assumption

**To Reproduce**
PJDFSTEST chown/05.t
```
expect 0 -u 65534 -g 65533,65534 -- chown ${n1}/${n2} -1 65533
expect 65534,65533 -u 65534 -g 65534 stat ${n1}/${n2} uid,gid
expected 65534,65533, got 65534,65534
```

**Expected behavior**
Chown should be able to set owner or group separately

**Urgency**
Describe the impact and urgency of the bug.

**Are you planning to fix it**
Please indicate if you are already working on a PR. 

**Additional context**

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?

Refactor user group policy and fixes #15822

### Why are the changes needed?

Support different user group authentication policy models.
Support set owner or group only

### CODE REVIEW

**Comment by yyongycy** [comment]
A few questions:
 1. Is it just a refactor or some logic change?
 2. Are there tests protecting?
 3. Do these pass related posix tests?

**Comment by LuQQiu** [comment]
@yyongycy it's not just refactor but includes logic changes
especially how to deal with the user group
Previously
- has a PropertyKey user.group.translation.enabled to decide whether to translate the user group in getattr(). And if it's false, all the file/group owner/group will be showed as default owner group and chown is not allowed.
- Has two user group policies used to set the file/dir owner/group when creating it. SystemUserGroupPolicy sets  the owner/group to end-user. CustomUserGroupPolicy sets the owner/group to a user set owner/group.
- Does not support chown owner only or chown group only.

Now
- Remove the user.group.translation.enabled from Alluxio JNI Fuse
- Have three user group policies
- - the default LaunchUserGroupPolicy: didn't do anything, use the Fuse launch user as the file/dir owner/group. get file/dir status always return the launch user/group
- - SystemUserGroupPolicy: use end-user owner/group as the file/dir owner/group. get file/dir status always do the user/group mapping and return the actual uid/gid (default to -1)
- - CustomerUserGroupPolicy: same as before
- Always allow chown command, and allows chown user only or group only

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioJnrFuseFileSystem.java [hunk 1]
```diff
@@ -96,8 +96,10 @@ public final class AlluxioJnrFuseFileSystem extends FuseStubFS
   @VisibleForTesting
   public static final long ID_NOT_SET_VALUE_UNSIGNED = 4294967295L;
 
-  private static final long UID = AlluxioFuseUtils.getUid(System.getProperty("user.name"));
-  private static final long GID = AlluxioFuseUtils.getGid(System.getProperty("user.name"));
+  private static final long UID = AlluxioFuseUtils.getUid(System.getProperty("user.name"))
```
Comment (yyongycy) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioJnrFuseFileSystem.java:99]: are we still using this wrapper?  If it is not used, maybe archive it?

File: integration/fuse/src/main/java/alluxio/fuse/auth/SystemUserGroupAuthPolicy.java [hunk 2]
```diff
@@ -13,100 +13,60 @@
...
 /**
- * Default Fuse Auth Policy.
+ * The system user group authentication policy that always set the user group
+ * to the actual end user.
+ * Note that this may downgrade the performance of creating file/directory.
  */
-public final class SystemUserGroupAuthPolicy implements AuthPolicy {
-  private static final Logger LOG = LoggerFactory.getLogger(
-      SystemUserGroupAuthPolicy.class);
+public final class SystemUserGroupAuthPolicy extends LaunchUserGroupAuthPolicy {
```
Comment (yyongycy) [inline integration/fuse/src/main/java/alluxio/fuse/auth/SystemUserGroupAuthPolicy.java:30]: this inheritance relation looks a little strange,

Comment (LuQQiu) [reply integration/fuse/src/main/java/alluxio/fuse/auth/SystemUserGroupAuthPolicy.java:30]: Yeah, it's more for reusing the code, the method in LaunchUserGroupAuthPolicy can be reused, i will think of whether there are better ways to do the same thing

**Review by yyongycy — APPROVED** [review]
LGTM

**Comment by LuQQiu** [comment]
@rongrong PTAL the code structure and code quality, thanks!

**Comment by LuQQiu** [comment]
The Alluxio POSIX API doc will be updated by updating the configuration chart directly through another PR to update the doc configuration.

**Comment by LuQQiu** [comment]
Error: 1.601 [ERROR] alluxio.util.JvmPauseMonitorTest.testMockedInfoPause  Time elapsed: 1.551 s  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at alluxio.util.JvmPauseMonitorTest.testMockedInfoPause(JvmPauseMonitorTest.java:106)

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**